libcloudforensics/providers/azure/internal/network.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Azure Networking functionality."""
+
+from typing import TYPE_CHECKING, Optional, Tuple, Any, Dict
+
+# pylint: disable=import-error
+from azure.mgmt import network
+from msrestazure import azure_exceptions
+# pylint: enable=import-error
+
+from libcloudforensics.providers.azure.internal import common
+
+if TYPE_CHECKING:
+  # TYPE_CHECKING is always False at runtime, therefore it is safe to ignore
+  # the following cyclic import, as it it only used for type hints
+  from libcloudforensics.providers.azure.internal import account  # pylint: disable=cyclic-import
+
+
+class AZNetwork:
+  """Azure Networking functionality.
+
+  Attributes:
+    az_account (AZAccount): An Azure account object.
+    network_client (NetworkManagementClient): An Azure network client object.
+  """
+
+  def __init__(self,
+               az_account: 'account.AZAccount') -> None:
+    """Initialize the Azure network class.
+
+    Args:
+      az_account (AZAccount): An Azure account object.
+    """
+    self.az_account = az_account
+    self.network_client = network.NetworkManagementClient(
+        self.az_account.credentials, self.az_account.subscription_id)
+
+  def CreateNetworkInterface(self,
+                             name: str,
+                             region: Optional[str] = None) -> str:
+    """Create a network interface and returns its ID.
+
+    Args:
+      name (str): The name of the network interface.
+      region (str): Optional. The region in which to create the network
+          interface. Default uses default_region of the AZAccount object.
+
+    Returns:
+      str: The id of the created network interface.
+
+    Raises:
+      ValueError: if name is not provided.
+      RuntimeError: If no network interface could be created.
+    """
+    if not name:
+      raise ValueError('name must be specified. Provided: {0!s}'.format(name))
+
+    if not region:
+      region = self.az_account.default_region
+
+    network_interface_name = '{0:s}-nic'.format(name)
+    ip_config_name = '{0:s}-ipconfig'.format(name)
+
+    # Check if the network interface already exists, and returns its ID if so.
+    try:
+      nic = self.network_client.network_interfaces.get(
+          self.az_account.default_resource_group_name, network_interface_name)
+      nic_id = nic.id  # type: str
+      return nic_id
+    except azure_exceptions.CloudError:
```

Comment:
Is it not possible that CloudError would also encapsulate other error situations than just "the NIC already exists"?
If something else goes wrong, we just eat the exception and continue..

Reply:
Changed to raise the exception if it's not about the NIC not being found (which is the one we wanna ignore)